index.js

```diff
@@ -93,121 +114,153 @@ rl.on('line', function(line) {
+            console.log('token for master user:', masterToken);
 
-function persistData(admin,
-                     users,
-                     groupName) {
-    return new Promise((resolve, reject) => { 
-        // Create admin user 
-        console.log('Creating admin user', admin.forename + '......');
-        var jsonData = JSON.stringify(admin);
-        request('POST', '/users', {'Content-Type': 'application/json',
-                           'Content-Length': jsonData.length}, jsonData).then(result => {
-            console.log('Admin user created with username:', result.username, 'id:', result.id);
-            var postData = {
-                username: admin.username,
-                password: admin.password
-            };
-        
-            // Login admin user
-            postData = JSON.stringify(postData);
-            console.log('Logging in admin user......');
-            request('POST', '/login', {'Content-Type': 'application/json',
-                                       'Content-Length': postData.length}, postData).then(result => {
-                console.log('Webtoken from login:', result.token);
-                var token = result.token;
-        
-                // Create group (admin user will be autmatically added to group)
-                var groupData = {
-                    name: groupName
-                };
-                console.log('Creating group ' + groupData.name + '.......');
-                jsonData = JSON.stringify(groupData);
-                request('POST', '/groups', {'Content-Type': 'application/json',
-                                    'Content-Length': jsonData.length,
-                                    'Authorization': 'Bearer ' + token}, jsonData).then(result => {
-                    var groupId = result.id;
-                    console.log('group created, id:', groupId);
-        
-                    function handleCreateUser(user) {
-                        // Create users
-                        var postData = JSON.stringify(user);
-                        console.log('Creating user ' + user.forename + '......');
-                        return request('POST', '/users', {'Content-Type': 'application/json',
-                                                          'Content-Length': postData.length}, postData);
-                    }
-                    console.log('Creating users......');
-                    var createUserPromises = users.map(handleCreateUser);
-                    var userIdMap = new Map();
-                    Promise.all(createUserPromises).then(results => {
-                        results.forEach(result => {
-                            userIdMap.set(result.username, result.id);
-                            console.log('User created with username:', result.username, 'id:', result.id);
-                        });
+            console.log('Creating admin user', admin.forename, '......');
+            createUser(admin, masterToken).then(result => {
```

Comment:
My understanding is when the admin user creates the group the admin user is automatically added to the group. In this case, the admin user doesn't need to be in the userIdMap, correct?